src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/GlobalFlowStateAnalysis/GlobalFlowStateAnalysisValueSet.cs

```diff
@@ -51,6 +51,7 @@ public GlobalFlowStateAnalysisValueSet(GlobalFlowStateAnalysisValueSet parent1,
                    height: Math.Max(parent1.Height, parent2.Height) + 1,
                    GlobalFlowStateAnalysisValueSetKind.Known)
         {
+            Debug.Assert(parent1 != parent2);
```

Comment:
Consider replacing public constrictors with static factory method, so the duplication handling can be done in a single location.